packages/ember-glimmer/lib/components/curly-component.js

```diff
@@ -66,6 +67,11 @@ class CurlyComponentManager {
       bucket.classRef = args.named.get('class');
     }
 
+    assert('You cannot use `classNameBindings` on a tag-less component: ' + component.toString(), () => {
```

Comment:
I'm guessing [`element`](http://emberjs.com/api/classes/Ember.Component.html#property_element) should not be allowed either